addons/account/models/account_move_line.py

```diff
@@ -1081,6 +1087,16 @@ def _compute_reconciled_lines_ids(self):
         for line in self:
             line.reconciled_lines_ids = line.matched_debit_ids.debit_move_id + line.matched_credit_ids.credit_move_id
 
+    @api.depends('matched_debit_ids', 'matched_credit_ids')
+    def _compute_reconciled_lines_excluding_exchange_diff_ids(self):
+        for line in self:
+            all_lines = line.matched_debit_ids.debit_move_id + line.matched_credit_ids.credit_move_id
+            excluded_ids = (
+                line.matched_debit_ids.exchange_move_id.line_ids.ids +
+                line.matched_credit_ids.exchange_move_id.line_ids.ids
+            )
+            line.reconciled_lines_excluding_exchange_diff_ids = all_lines.filtered(lambda l: l.id not in excluded_ids)
```

Comment:
In fact we don't have to filtered you can just do all_lines - excluded_lines if you remove the ".ids" from the excluded lines to use the record set :smile: